compaction/compaction.cc

```diff
@@ -1212,18 +1212,57 @@ class cleanup_compaction final : public regular_compaction {
         // optimize set of potentially overlapping ranges by deoverlapping them.
         non_owned_ranges = dht::partition_range::deoverlap(std::move(non_owned_ranges), dht::ring_position_comparator(*_schema));
 
```

Comment:
I think we need to document dht::partition_range::deoverlap() (above the function declaration) returns a sorted vector, such that no change in the future breaks this assumption.